### PR TITLE
Frontend Structure

### PR DESCRIPTION
**Pull request by OmerRosenbaum**
:link: Link to Swimm doc: [Frontend Structure](https://app.swimm.io/workspaces/aRvMqc0yWAVcJlLN944D/repos/Z2l0aHViJTNBJTNBc2VudHJ5JTNBJTNBc3dpbW1pbw==/branch/frontend-structure-swimm-doc-jcliy/docs/c1t7o).

---
  

  Have any questions? Check out [Swimm docs](https://docs.swimm.io/) and [join our Slack Community](https://swimm.io/slack).

### CODE REVIEW

File: .swm/frontend-structure.c1t7o.sw.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+*   `📄 static/app/actionCreators` - action creator functions are used to dispatch actions. For example, `📄 static/app/actionCreators/group.tsx`. Reflux stores listen to actions and update themselves accordingly.
+    
+
+To learn more about our use of Reflux, refer to [State Management with Reflux](state-management-with-reflux.jni2e.sw.md) .
+
+# Important Files
+
+## `📄 static/app/main.tsx`
+
+This is the web app, it is intended to be available online.
```
Comment (saar-swimm) [inline .swm/frontend-structure.c1t7o.sw.md:63]: How is one file 'the web app'? Maybe 'the web app entrypoint'? also the 'intented to be available online sentence is unclear to me

File: .swm/state-management-with-reflux.jni2e.sw.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+# 1\. Creating Actions
+
+An action is a [function object](http://en.wikipedia.org/wiki/Function_object) that can be invoked like any other function.
+
+Actions are registered under `📄 static/app/actions` , e.g., `PageFiltersActions`[<sup id="Z2wU2DS">↓</sup>](#f-Z2wU2DS) are registered in `📄 static/app/actions/pageFiltersActions.tsx`
+
+<br/>
+
+To create actions we use `createActions`[<sup id="1Hy98y">↓</sup>](#f-1Hy98y) , like so:
```
Comment (saar-swimm) [inline .swm/state-management-with-reflux.jni2e.sw.md:43]: Unclear to me if this creates all the actions? only a certain group of actions? are there other calls to `createActions?`

File: .swm/state-management-with-reflux.jni2e.sw.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+⬜ 92      * Reset values in the page filters store
+⬜ 93      */
+🟩 94     export function resetPageFilters() {
+🟩 95       PageFiltersActions.reset();
+🟩 96     }
+```
+
+<br/>
+
+Reflux stores listen to actions and update themselves accordingly.
```
Comment (saar-swimm) [inline .swm/state-management-with-reflux.jni2e.sw.md:154]: How can a store listen to an action?

**Review by saar-swimm — CHANGES_REQUESTED** [review]

File: .swm/state-management-with-reflux.jni2e.sw.md (hunk 2 above)
Comment (OmerRosenbaum) [reply .swm/state-management-with-reflux.jni2e.sw.md:43]: I actually think this sentence is straightforward. How would you rewrite it?
(It's for all actions, and there are many calls...)

File: .swm/state-management-with-reflux.jni2e.sw.md (hunk 3 above)
Comment (OmerRosenbaum) [reply .swm/state-management-with-reflux.jni2e.sw.md]: I clarified

File: .swm/frontend-structure.c1t7o.sw.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  static/app/sentryTypes.tsx: 0fe070ac5d22d1d33a58de460c6c56b3b9841abf
+---
+
+The frontend codebase is under `📄 static/app` . Our framework is [ReactJS](https://reactjs.org/). There are additional folders related to the frontend in additional locations, as described below.
+
+The main folders to know are:
+
+# Components
+
+We use components for UI that are designed to be highly reusable.
```
Comment (rotembar3) [inline .swm/frontend-structure.c1t7o.sw.md:16]: Components for UI -> UI components

File: .swm/state-management-with-reflux.jni2e.sw.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+```
++---------+       +--------+       +-----------------+
+¦ Actions ¦------>¦ Stores ¦------>¦ View Components ¦
++---------+       +--------+       +-----------------+
+     ^                                      ¦
+     +--------------------------------------+
+```
+
+The pattern is composed of actions and data stores, where actions initiate new data to pass through data stores before coming back to the view components again. If a view component has an event that needs to make a change in the application's data stores, they need to do so by signaling to the stores through the actions available.
```
Comment (rotembar3) [inline .swm/state-management-with-reflux.jni2e.sw.md:27]: In this pattern, actions initiate the process of data passing through data stores before returning to the view components. 

I feel like this is a bit clearer, wdyt?

Comment (rotembar3) [reply .swm/state-management-with-reflux.jni2e.sw.md:27]: A view component's event that requires changes to the application's data stores signals the stores with the available actions.

**Review by rotembar3 — APPROVED** [review]

**Pull request merged**